### PR TITLE
[Scheduler] Fix missing job in scheduler error message

### PR DESCRIPTION
**Pull request by AlonMaor14**
Raise more informative error when trying to update a schedule that doesn't exist in the `APScheduler` memory

### CODE REVIEW

File: mlrun/api/utils/scheduler.py [hunk 1]
```diff
@@ -640,15 +641,39 @@ def _update_schedule_in_scheduler(
...
+                func=function,
+                args=args,
+                kwargs=kwargs,
+                trigger=trigger,
+                next_run_time=next_run_time,
+            )
+        except JobLookupError as exc:
+            raise mlrun.errors.MLRunNotFoundError(
+                f"Schedule job with id {job_id} not found in scheduler. Reload is required."
+            ) from exc
```
Comment (liranbg) [inline mlrun/api/utils/scheduler.py:675]: reload what?

File: tests/api/utils/test_scheduler.py [hunk 2]
```diff
@@ -1155,6 +1157,39 @@ async def test_update_schedule_failure_not_found(db: Session, scheduler: Schedul
...
+        schedule_name,
+        schemas.ScheduleKinds.job,
+        scheduled_object,
+        inactive_cron_trigger,
+        1,
+    )
+
+    with pytest.raises(mlrun.errors.MLRunNotFoundError) as excinfo:
+        scheduler.update_schedule(
+            db, mlrun.api.schemas.AuthInfo(), project_name, schedule_name
```
Comment (liranbg) [inline tests/api/utils/test_scheduler.py:1184]: can you comment _why_ it would raise?

File: tests/api/utils/test_scheduler.py [hunk 3]
```diff
@@ -1155,6 +1157,39 @@ async def test_update_schedule_failure_not_found(db: Session, scheduler: Schedul
...
+        )
+    job_id = scheduler._resolve_job_id(project_name, schedule_name)
+    assert (
+        f"Schedule job with id {job_id} not found in scheduler. Reload is required."
+        in str(excinfo.value)
+    )
```
Comment (liranbg) [inline tests/api/utils/test_scheduler.py:1190]: can you _reload_ and shows it works as expected?

**Review by liranbg — CHANGES_REQUESTED** [review]
Minor comments

File: tests/api/utils/test_scheduler.py (hunk 3 above)
Comment (AlonMaor14) [reply tests/api/utils/test_scheduler.py:1190]: Currently, reload is done only by restarting the API chief (or service).
We need to add an endpoint to trigger reload manually  - we'll need to discuss it offline.
We had this issue on a live system and restarting the chief fixed it.

File: mlrun/api/utils/scheduler.py (hunk 1 above)
Comment (AlonMaor14) [reply mlrun/api/utils/scheduler.py:675]: changed the error msg

**Review by quaark — APPROVED** [review]

**Pull request merged**